### PR TITLE
Introduce resolvers for api events

### PR DESCRIPTION
**Pull request by alextargov**
**Description**

Changes proposed in this pull request:
- Add resolvers for Listing in pages, Creating and Deleting APIs in the context of an Application.
- Add resolvers for Listing in pages, Creating and Deleting Events in the context of an Application.

**Related issue(s)**

- ...

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [x] Mocks are regenerated, using the automated script

### CODE REVIEW

**Comment by alextargov** [comment]
reset-vm

**Comment by alextargov** [comment]
reset-vm

File: components/director/internal/domain/api/repository.go [hunk 1]
```diff
@@ -137,6 +137,27 @@ func (r *pgRepository) ListByApplicationID(ctx context.Context, tenantID, appID
 	return apis, nil
 }
 
+// ListByApplicationIDPage lists all APIDefinitions for a given application ID with paging.
+func (r *pgRepository) ListByApplicationIDPage(ctx context.Context, tenantID string, appID string, pageSize int, cursor string) (*model.APIDefinitionPage, error) {
+	var apiDefCollection APIDefCollection
+	page, totalCount, err := r.pageableQuerier.List(ctx, resource.API, tenantID, pageSize, cursor, "id", &apiDefCollection, repo.NewEqualCondition("app_id", appID))
```
Comment (dragobt) [inline components/director/internal/domain/api/repository.go:143]: use idColumn var

File: components/director/internal/domain/eventdef/repository.go [hunk 2]
```diff
@@ -93,6 +95,28 @@ func (r *pgRepository) GetForBundle(ctx context.Context, tenant string, id strin
 	return r.GetByID(ctx, tenant, id)
 }
 
+// ListByApplicationIDPage lists all APIDefinitions for a given application ID with paging.
+func (r *pgRepository) ListByApplicationIDPage(ctx context.Context, tenantID string, appID string, pageSize int, cursor string) (*model.EventDefinitionPage, error) {
+	var apiDefCollection EventAPIDefCollection
+	page, totalCount, err := r.pageableQuerier.List(ctx, resource.EventDefinition, tenantID, pageSize, cursor, "id", &apiDefCollection, repo.NewEqualCondition("app_id", appID))
```
Comment (dragobt) [inline components/director/internal/domain/eventdef/repository.go:101]: use idColumn and appColumn vars

File: components/director/internal/domain/eventdef/resolver.go [hunk 3]
```diff
@@ -74,6 +81,61 @@ func NewResolver(transact persistence.Transactioner, svc EventDefService, bndlSv
 	}
 }
 
+// EventDefinitionsForApplication lists all APIDefinitions for a given application ID with paging.
```
Comment (dragobt) [inline components/director/internal/domain/eventdef/resolver.go:84]: `APIDefinitions` -> `eventDefinitions`

File: components/director/internal/domain/eventdef/service.go [hunk 4]
```diff
@@ -99,6 +104,20 @@ func (s *service) ListByApplicationID(ctx context.Context, appID string) ([]*mod
 	return s.eventAPIRepo.ListByApplicationID(ctx, tnt, appID)
 }
 
+// ListByApplicationIDPage lists all APIDefinitions for a given application ID with paging.
```
Comment (dragobt) [inline components/director/internal/domain/eventdef/service.go:107]: `APIDefinitions` -> `eventDefinitions`

File: components/director/internal/domain/eventdef/service.go [hunk 5]
```diff
@@ -232,17 +256,32 @@ func (s *service) UpdateInManyBundles(ctx context.Context, id string, in model.E
...
+
+	if err = s.eventAPIRepo.Update(ctx, tnt, event); err != nil {
+		return errors.Wrapf(err, "while updating EventDefinition with id %s", id)
+	}
+
+	if specIn != nil {
+		return s.handleSpecsInEvent(ctx, id, specIn)
 	}
 
 	return nil
```
Comment (dragobt) [inline components/director/internal/domain/eventdef/service.go:287]: not tested

File: components/director/internal/domain/root_resolver.go [hunk 6]
```diff
@@ -449,6 +449,14 @@ func (r *queryResolver) Viewer(ctx context.Context) (*graphql.Viewer, error) {
 	return r.viewer.Viewer(ctx)
 }
 
+func (r *queryResolver) ApisForApplication(ctx context.Context, appID string, first *int, after *graphql.PageCursor) (*graphql.APIDefinitionPage, error) {
```
Comment (dragobt) [inline components/director/internal/domain/root_resolver.go:452]: add go docs to all introduced functions in this file

File: tests/director/tests/event_api_def_test.go [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	"github.com/kyma-incubator/compass/tests/pkg/testctx"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEventDefinitionInApplication(t *testing.T) {
+	ctx := context.Background()
+
+	tenantId := tenant.TestTenants.GetDefaultTenantID()
+	appName := "app-test-bundle"
```
Comment (dragobt) [inline tests/director/tests/event_api_def_test.go:19]: i am not sure why in all introduced tests `appName` contain the word bundle

File: tests/director/tests/event_api_def_test.go [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+	assert.NotNil(t, eventDef.ID)
+	saveExample(t, eventAddRequest.Query(), "add event definition to application")
+}
+
+func TestUpdateEventDefinitionToApplication(t *testing.T) {
+	ctx := context.Background()
+
+	appName := "app-test-bundle"
+	newName := "TestUpdatePIDefinitionToApplication"
```
Comment (dragobt) [inline tests/director/tests/event_api_def_test.go:66]: typo "API"

File: tests/pkg/fixtures/event_api_definition_queries.go [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+	request := FixAddEventToApplicationRequest(appID, inputGQL)
+	eventDef := graphql.EventAPIDefinitionExt{}
+
+	err = testctx.Tc.RunOperationWithCustomTenant(ctx, gqlClient, tenant, request, &eventDef)
+	require.NoError(t, err)
+	return eventDef
+}
+
+func AddEventToApplication(t require.TestingT, ctx context.Context, gqlClient *gcli.Client, appID string) graphql.EventAPIDefinitionExt {
+	return AddEventToApplicationWithInput(t, ctx, gqlClient, tenant.TestTenants.GetDefaultTenantID(), appID, FixEventAPIDefinitionInput())
```
Comment (dragobt) [inline tests/pkg/fixtures/event_api_definition_queries.go:26]: maybe this should be `FixEventAPIDefinitionInput()` -> `FixEventDefinitionInput()`

**Review by dragobt — APPROVED** [review]

**Review by nkanchev — DISMISSED** [review]

**Review by la4ezar — APPROVED** [review]

**Review by la4ezar — APPROVED** [review]

**Review by dragobt — APPROVED** [review]

**Review by nkanchev — APPROVED** [review]

**Review by petar-iv — APPROVED** [review]

**Pull request merged**